reposerver/src/test/scala/com/advancedtelematic/tuf/reposerver/util/ResourceSpec.scala

```diff
@@ -62,32 +62,37 @@ class FakeKeyserverClient(defaultKeyType: KeyType) extends KeyserverClient {
     pendingRequests.put(repoId, true)
   }
 
-  private lazy val preGeneratedKeys = RoleType.ALL.map { role =>
-    val TufKeyPair(publicKey, privateKey) = TufCrypto.generateKeyPair(defaultKeyType, defaultKeyType.crypto.defaultKeySize)
-    role -> new KeyPair(publicKey.keyval, privateKey.keyval)
+  // exclude root since we don't know the corresponding KeyType yet
+  private lazy val preGeneratedKeys: Map[RoleType, TufKeyPair] = RoleType.ALL.filter(_ != RoleType.ROOT).map { role =>
+    role -> TufCrypto.generateKeyPair(defaultKeyType, defaultKeyType.crypto.defaultKeySize)
   }.toMap
```

Comment:
what do you mean? A root key type should determine the default for the other key types.

Reply:
That would mean we can't pre-generate keys, which would also make things easier.

Speaking of which, I'm assuming that only the root key is generated on the server, is that true?

Reply:
all the base keys are generated on keyserver, not just root. the user can then replace those keys with offline keys generated through cli